paasta_tools/deployd/common.py

```diff
@@ -1,31 +1,61 @@
 import logging
 import time
 from collections import namedtuple
+from queue import PriorityQueue
 from queue import Queue
 from threading import Thread
 
 from paasta_tools.marathon_tools import DEFAULT_SOA_DIR
 from paasta_tools.marathon_tools import get_all_marathon_apps
 from paasta_tools.marathon_tools import get_marathon_client
 from paasta_tools.marathon_tools import load_marathon_config
+from paasta_tools.marathon_tools import load_marathon_service_config
 from paasta_tools.marathon_tools import load_marathon_service_config_no_cache
 from paasta_tools.utils import InvalidJobNameError
 from paasta_tools.utils import NoDeploymentsAvailable
 from paasta_tools.utils import NoDockerImageError
 
 BounceTimers = namedtuple('BounceTimers', ['processed_by_worker', 'setup_marathon', 'bounce_length'])
-ServiceInstance = namedtuple(
+ServiceInstanceTuple = namedtuple(
     'ServiceInstance', [
         'service',
         'instance',
         'bounce_by',
         'watcher',
         'bounce_timers',
         'failures',
+        'priority',
     ],
 )
 
 
+def ServiceInstance(service, instance, watcher, cluster, bounce_by, failures=0, bounce_timers=None, priority=None):
+    if priority is None:
+        priority = get_priority(service, instance, cluster)
+    return ServiceInstanceTuple(
+        service=service,
+        instance=instance,
+        watcher=watcher,
+        bounce_by=bounce_by,
+        failures=failures,
+        bounce_timers=bounce_timers,
+        priority=priority,
+    )
+
+
+def get_priority(service, instance, cluster):
```

Comment:
I think I prefer it here. I was a bit worried that we will read the config every time we create a `ServiceInstance` but there is a 5s cache so I think it'll be okay? We do read the config in some cases when we create the `ServiceInstance` but even then we don't expose it cleanly. I feel like this might be the least messy way to do it.